### ISSUE TITLE
skeletonize 3D with different voxel size

### ISSUE THREAD

**Issue by modaresimr** [scikit-image/scikit-image#6512]
## Description
Based on the description on [here](https://scikit-image.org/docs/stable/api/skimage.morphology.html), the `skeletonize` or `medial_axis` functions cannot compute the skeleton based on the voxel size. For example, in CT scan the thickness of z axis can be 5mm while the other axis can be 1 mm. In sk-image library it is always considered in other functions like distance_transform_edt with the scaling parameter, however, it is not considered in morphology.

**Comment by lagru** [comment]
Correct, it seems to be the case for `skeletonize` that different the physical voxel size is not taken into account. Are you asking for this feature?

**Comment by modaresimr** [comment]
@lagru Yes, I am asking for this feature

**Comment by lagru** [comment]
Okay, noted. Are you perhaps familiar enough with the algorithm to give it a try yourself? :grin: Otherwise this will likely have to wait until someone else finds the time, need and knowledge to tackle this. 

In the meantime, it's good to have this issue open to document the feature request. :slightly_smiling_face:

**Comment by modaresimr** [comment]
I will try to look at it, 
I am adding the one example of the effect of pixel/voxel size.
the pixel size in the left image is (1,1) but in the right image is (1,2). Infact, they are the same image, therefore, their skeleton should be the same. 
![image](https://user-images.githubusercontent.com/9498182/189542909-19f99291-0510-4ffb-ba0f-3ab2bed2005f.png)


code to produce:
```
import matplotlib.pyplot as plt
from skimage import draw
arr = np.zeros((20, 20))
rr, cc = draw.ellipse(10, 10, 8, 8, arr.shape)
arr[rr, cc] = 1
arr+=skeletonize(arr)
fig,axes=plt.subplots(1,2)
axes[0].imshow(arr)

arr = np.zeros((20, 10))
rr, cc = draw.ellipse(10, 5, 8, 4, arr.shape)
arr[rr, cc] = 1
arr+=skeletonize(arr)
axes[1].imshow(arr)
```

**Comment by grlee77** [comment]
In terms of the numerical `distance` returned by the `return_distance` option for `medial_axis`, it should just be a matter of adding a `spacing=None` kwarg to the API and then passing that along to the internal call to `distance_transform_edt`. That does not seem like it would resolve the difference in the shape of the skeleton you showed above though. Have you found any other implementations that support inisotropic spacing and would return the same skeleton in that case? (maybe something in SimpleITK? at least I know ITK typically takes pixel/voxel sizes into account in its algorithms)

**Comment by modaresimr** [comment]
Good i will do it and let you know

**Comment by modaresimr** [comment]
It works perfectly for 2D images, however for 3D images it is not working .
I will create a pull request for adding spacing in media axis. 
and work more on making it available for 3D images.

**Comment by modaresimr** [comment]
unfortunately, it won't work for 3d
even that i have add all conditions in Lee94 but it won't work well

**Comment by modaresimr** [comment]
I have tried to upgrade the  `skeletonize_3d` for different voxel size. But it won't work correctly.
I am writing my idea so everyone can help to resolve the issue.

# Current Algorithm (short description)
`
from border in (W, E, S, N, U, B):
    shrink one voxel in border if it doesn't create new hole and does not break connectivity
`
# my idea
My idea is to shrink the border inverse to the voxel size. e.g., if voxel size is dx,dy,dz=(1,2,3), then call_count is 3,1.5,1
`
allowed_call={'S': max(voxelsize)/voxelsize[0], 'N': max(voxelsize)/voxelsize[0], 
                    'W': max(voxelsize)/voxelsize[1], 'E': max(voxelsize)/voxelsize[1], 
                    'U': max(voxelsize)/voxelsize[2], 'B': max(voxelsize)/voxelsize[2]}
from border in (W, E, S, N, U, B):
    if is_allowed(border):
         shrink one voxel in border if it doesn't create new hole and does not break connectivity
`
It means that it will call shrink 6 times for S and N, 3 times for W and E and 2 times for U and B
